### PR TITLE
Gen yaml annotations for flow proto

### PR DESCRIPTION
**Pull request by marqc**
Please ensure your pull request adheres to the following guidelines:

- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.

Add yaml annotations to generated flow proto as discussed in https://github.com/cilium/cilium/pull/28873#discussion_r1377689955
Requires: https://github.com/cilium/cilium/pull/28943

cc: @chancez @glibsm 

```release-note
Add yaml annotations to generated flow proto.
```

### CODE REVIEW

File: api/v1/Makefile.protoc
Comment (gandro) [inline api/v1/Makefile.protoc]: This looks incredibly dangerous. Anyone running this accidentally outside the build container is potentially going to loose data.

Can we instead fix up the mount path in the `Makefile`?

**Review by gandro — CHANGES_REQUESTED** [review]
Thanks. The overall idea looks good to me, but I'm not a big fan of deleting whole folders during the build.

File: api/v1/Makefile
Comment (marqc) [inline api/v1/Makefile]: fix once the image is built from https://github.com/cilium/cilium/pull/28943

File: api/v1/Makefile.protoc
Comment (marqc) [reply api/v1/Makefile.protoc]: fixed

File: api/v1/Makefile.protoc
Comment (gandro) [inline api/v1/Makefile.protoc]: How does this work without passing in all the plugin parameters? Are protoc and the plugins ignoring any already present files?

Or asked differently - why do we need to pass in `$(HUBBLE_PROTOC_PLUGINS)` here? Wouldn't the following be sufficient?
```suggestion
		$(PROTOC) --plugin=$(GOPATH)/bin/protoc-gen-gotag -I $(HUBBLE_PROTO_PATH) \
```

On a side-note: I liked that the previous version had a commit why this is needed. I think it would be valuable to have a comment here that `gotag` needs the file to already exist

Comment (marqc) [reply api/v1/Makefile.protoc]: You're right, only gotag plugin is needed in the second invocation. I changed it and also rebased the commit to include required image changes in this PR

File: api/v1/Makefile.protoc
Comment (glibsm) [inline api/v1/Makefile.protoc:39]: Can this plugin be included in $HUBBLE_PROTOC_PLUGINS and run as part of the previous command?

Comment (marqc) [reply api/v1/Makefile.protoc:39]: I tried that, but it fails then with an error of multiple attempts to write to the same file. 
I believe it is mentioned on the plugin page as a known limitation https://github.com/srikrsna/protoc-gen-gotag#note

**Comment by chancez** [comment]
LGTM. I'll approve after https://github.com/cilium/cilium/pull/28943 is merged and the base image + CI passes.

**Review by gandro — APPROVED** [review]

**Review by christarazi — COMMENTED** [review]
Can you add justification in the commit msgs for why the changes are needed?

**Comment by marqc** [comment]
> Can you add justification in the commit msgs for why the changes are needed?

@christarazi Added.

**Review by christarazi — APPROVED** [review]

**Review by bimmlerd — APPROVED** [review]
Fine by me, but will unfortunately conflict with https://github.com/cilium/cilium/pull/28945. Given both are currently blocked by the Cloudflare outage, let's just see which one gets in first and rebase the other one.

**Comment by marqc** [comment]
> Fine by me, but will unfortunately conflict with #28945. Given both are currently blocked by the Cloudflare outage, let's just see which one gets in first and rebase the other one.

Rebased.

**Comment by gandro** [comment]
@marqc Can you rebase again. Unfortunately `main` was broken due to a merge race (c.f. https://github.com/cilium/cilium/pull/28745). Thanks!

**Comment by squeed** [comment]
Travis took a nap :-(. If this PR doesn't need any more pushes, then a close-and-reopen will do the trick.

**Comment by marqc** [comment]
I just realized that names generated with this plugin are inconsitent with json tags. This would make a mess in the API. I will close this and instead I will use different YAML parser that works with json tags  `sigs.k8s.io/yaml`

**Pull request closed without merging**
